### ISSUE TITLE
GH issue updater is broken

### ISSUE THREAD

**Issue by jerboaa** [graalvm/mandrel#687]
## Description
I was manually looking through CI and noticed that there are some failures which should be reported automatically by opening an issue. However that doesn't seem to work (for a while now):

```
 The CI build had status failure.

Getting logs for job Q main M 24.0 JDK 22 / Set distribution, build-from-source, and maven-deploy-local variables based on build-type
Unable to get logs for job Q main M 24.0 JDK 22 / Set distribution, build-from-source, and maven-deploy-local variables based on build-type
org.kohsuke.github.HttpException: ﻿<?xml version="1.0" encoding="utf-8"?>
<Error><Code>AuthenticationFailed</Code><Message>Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
RequestId:2170685c-101e-00b4-68c4-6a97a5000000
Time:2024-02-29T04:05:22.7917907Z</Message></Error>
	at org.kohsuke.github.GitHubConnectorResponseErrorHandler$1.onError(GitHubConnectorResponseErrorHandler.java:62)
	at org.kohsuke.github.GitHubClient.detectKnownErrors(GitHubClient.java:504)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:464)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:427)
	at org.kohsuke.github.Requester.fetchStream(Requester.java:131)
	at org.kohsuke.github.GHWorkflowJob.downloadLogs(GHWorkflowJob.java:232)
	at Report.getJobsLogs(quarkus-ecosystem-issue.java:329)
	at Report.processLogs(quarkus-ecosystem-issue.java:233)
```

See:
https://github.com/graalvm/mandrel/actions/runs/8090848687/job/22108990298

The last time this worked was here (Feb 22, 2024):
https://github.com/graalvm/mandrel/actions/runs/7999237051/job/21846716216

@zakkak PTAL when you're back.

**Comment by jerboaa** [comment]
@zakkak Could you please take a look at this? Thanks!

**Comment by zakkak** [comment]
My so far understanding is that this is an issue in `org.kohsuke:github-api` triggered by a recent change in the location where the run logs get uploaded (I inferred this from this issue https://github.com/hub4j/github-api/issues/1790 which is related to `actions/upload-artifact@v4` and states that v4 in contrast to v3 uploads artifacts to a different infrastructure). There is an open PR aiming to address this issue https://github.com/hub4j/github-api/pull/1791 which seems to be at the final stages of approval and merging. I guess there is not much we can do on our side for the time being other than updating the script so that it fails when an exception is thrown to make it easier to detect such issues in the future...

**Comment by jerboaa** [comment]
Thanks for the analysis!

**Comment by jerboaa** [comment]
It looks like by moving to the upload-artifact@v4 version we could have this fixed. See https://github.com/hub4j/github-api/pull/1791#issuecomment-2001921075 I'll test a PR.

**Comment by zakkak** [comment]
> It looks like by moving to the upload-artifact@v4 version we could have this fixed. See [hub4j/github-api#1791 (comment)](https://github.com/hub4j/github-api/pull/1791#issuecomment-2001921075) I'll test a PR.

FTR that's not true. Moving to `upload-artifact@v4` would not have any impact in our case, since the logs that the GH issue updater fetches are not uploaded by us using `upload-artifact`. Furthermore, updating to `upload-artifact@v4` was what lead the Quarkus team to observe the issue, so even if we were manually uploading the logs using it in order to move to v4 we would need the github-api fix first.

**Comment by jerboaa** [comment]
> > It looks like by moving to the upload-artifact@v4 version we could have this fixed. See [hub4j/github-api#1791 (comment)](https://github.com/hub4j/github-api/pull/1791#issuecomment-2001921075) I'll test a PR.
> 
> FTR that's not true. Moving to `upload-artifact@v4` would not have any impact in our case, since the logs that the GH issue updater fetches are not uploaded by us using `upload-artifact`. Furthermore, updating to `upload-artifact@v4` was what lead the Quarkus team to observe the issue, so even if we were manually uploading the logs using it in order to move to v4 we would need the github-api fix first.

Yes, I realized that later. Thanks! The real fix was #702 (bumping the github-api to the fixed version). Either way, we needed the action updates to get rid of the Node 16 warnings.
